### PR TITLE
added requestbody for volume start api

### LINKED ISSUE #759: Volume start force failing, after brick killing
-> Kill any one brick from volume using kill -9 command,
-> Then perform  './glustercli volume start rep1 force '
-> volume start force failing with below error.

./glustercli volume start rep1 force
volume start failed

Error: Request failed with HTTP Status code 400

-> volume start force not able to start the killed brick.

Logs:

DEBU[2018-05-16 19:29:05.275846] attempting to lock                            key=locks/rep1 reqid=9ae04b94-9c9e-47be-a3ac-ee488e8cce1a source="[lock.go:109:transaction.CreateLockFuncs.func1]"
DEBU[2018-05-16 19:29:05.278693] lock obtained                                 key=locks/rep1 reqid=9ae04b94-9c9e-47be-a3ac-ee488e8cce1a source="[lock.go:113:transaction.CreateLockFuncs.func1]"
DEBU[2018-05-16 19:29:05.279700] attempting to unlock                          key=locks/rep1 reqid=9ae04b94-9c9e-47be-a3ac-ee488e8cce1a source="[lock.go:126:transaction.CreateLockFuncs.func2]"
DEBU[2018-05-16 19:29:05.281112] lock unlocked                                 key=locks/rep1 reqid=9ae04b94-9c9e-47be-a3ac-ee488e8cce1a source="[lock.go:132:transaction.CreateLockFuncs.func2]"
INFO[2018-05-16 19:29:05.281180] ::1 - - [16/May/2018:19:29:05 +0530] "POST /v1/volumes/rep1/start HTTP/1.1" 400 59  reqid=9ae04b94-9c9e-47be-a3ac-ee488e8cce1a

### PR DESCRIPTION
**Pull request by Madhu-1**
Fixes #759 
Signed-off-by: Madhu Rajanna <mrajanna@redhat.com>

### CODE REVIEW

**Comment by Madhu-1** [comment]
@prashanthpai  @aravindavk please review

File: glusterd2/commands/volumes/volume-start.go [hunk 1]
```diff
@@ -69,6 +73,12 @@ func volumeStartHandler(w http.ResponseWriter, r *http.Request) {
 	logger := gdctx.GetReqLogger(ctx)
 	volname := mux.Vars(r)["volname"]
 
+	query := r.URL.Query().Get("force")
```
Comment (prashanthpai) [inline glusterd2/commands/volumes/volume-start.go]: Name this variable as `force`. There can be many query params.

File: glusterd2/commands/volumes/volume-start.go [hunk 2]
```diff
@@ -69,6 +73,12 @@ func volumeStartHandler(w http.ResponseWriter, r *http.Request) {
...
+	force, err := strconv.ParseBool(query)
+	if err != nil {
+		restutils.SendHTTPError(ctx, w, http.StatusBadRequest, errors.ErrInvalidQuery)
```
Comment (prashanthpai) [inline glusterd2/commands/volumes/volume-start.go]: `ErrInvalidQuery` is very generic and not useful. Do not define a constant for that.

Instead, pass error string here in the handler.

```go
restutils.SendHTTPError(ctx, w, http.StatusBadRequest, "The query param force only accepts boolean values")
```

File: pkg/errors/error.go [hunk 3]
```diff
@@ -41,4 +41,5 @@ var (
 	ErrClusterNotFound         = errors.New("Cluster instance not found in store")
 	ErrDuplicateBrickPath      = errors.New("Duplicate brick entry")
 	ErrVolFileNotFound         = errors.New("volume file not found")
+	ErrInvalidQuery            = errors.New("Invalid query paramater")
```
Comment (prashanthpai) [inline pkg/errors/error.go]: Remove this. Return a specific error as commented above.

File: glusterd2/commands/volumes/volume-start.go (hunk 1 above)
Comment (Madhu-1) [reply glusterd2/commands/volumes/volume-start.go]: done

File: glusterd2/commands/volumes/volume-start.go (hunk 2 above)
Comment (Madhu-1) [reply glusterd2/commands/volumes/volume-start.go]: done

File: pkg/errors/error.go (hunk 3 above)
Comment (Madhu-1) [reply pkg/errors/error.go]: done

**Comment by prashanthpai** [comment]
retest this please

**Comment by Madhu-1** [comment]
@prashanthpai ,i will add it later

**Comment by aravindavk** [comment]
Since this is `POST` request, I prefer `force` can be sent as request body instead of query params. Query params are suitable for GET requests.

**Comment by prashanthpai** [comment]
> Since this is POST request, I prefer force can be sent as request body instead of query params. Query params are suitable for GET requests.

I looked this up and indeed it seems [semantically proper](https://stackoverflow.com/questions/14551194/how-are-parameters-sent-in-an-http-post-request/26717908#26717908) to avoid providing query params for POST.

@Madhu-1 Can you please change it as suggested ?

**Comment by Madhu-1** [comment]
@aravindavk  please review

File: pkg/restclient/volume.go [hunk 4]
```diff
@@ -90,9 +90,9 @@ func (c *Client) VolumeStatus(volname string) (api.VolumeStatusResp, error) {
 }
 
 // VolumeStart starts a Gluster Volume
-func (c *Client) VolumeStart(volname string) error {
+func (c *Client) VolumeStart(volname string, req api.VolumeStartReq) error {
```
Comment (aravindavk) [inline pkg/restclient/volume.go]: Second argument can be `force bool`, request can be formed inside this func.

Comment (Madhu-1) [reply pkg/restclient/volume.go]: @aravindavk done.

File: pkg/restclient/volume.go [hunk 5]
```diff
@@ -90,9 +90,12 @@ func (c *Client) VolumeStatus(volname string) (api.VolumeStatusResp, error) {
 }
 
 // VolumeStart starts a Gluster Volume
-func (c *Client) VolumeStart(volname string) error {
+func (c *Client) VolumeStart(volname string, force bool) error {
+	req := api.VolumeStartReq{
+		Force: force,
```
Comment (prashanthpai) [inline pkg/restclient/volume.go]: `Force` is too broad. Now that this in request body, we can name it more specifically. Something like - `ForceStartBricks` (var name) and `force-start-bricks` (in json)

Comment (Madhu-1) [reply pkg/restclient/volume.go]: @prashanthpai done

**Pull request merged**